### PR TITLE
set godo client user agent

### PR DESCRIPTION
**Pull request by bhcleek**
Set the user agent instead of relying on the default User-Agent string provided by godo.

### CODE REVIEW

File: builder/digitalocean/builder.go [hunk 1]
```diff
@@ -48,6 +50,8 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 	client := godo.NewClient(oauth2.NewClient(context.TODO(), &apiTokenSource{
 		AccessToken: b.config.APIToken,
 	}))
+	client.UserAgent = useragent.String(version.PluginVersion.FormattedVersion())
```
Comment (nywilken) [inline builder/digitalocean/builder.go:53]: I can see the need for having a more specific UserAgent over the default [godo UserAgent](https://github.com/digitalocean/godo/blob/v1.78.0/godo.go#L25). This looks good to me. 

Before I merge can you update the UserAgent for the client used in the [post-processor](https://github.com/hashicorp/packer-plugin-digitalocean/blob/c10f794e4005eec0cd9628ada13524b47b2a22c8/post-processor/digitalocean-import/post-processor.go#L202), as well.

Comment (nywilken) [inline builder/digitalocean/builder.go:53]: I can see the need for having a more specific UserAgent over the default [godo UserAgent](https://github.com/digitalocean/godo/blob/v1.78.0/godo.go#L25). Overall the change looks good to me. But the same change is needed for the post-processor. 

Can you please update the UserAgent for the client used in the [post-processor](https://github.com/hashicorp/packer-plugin-digitalocean/blob/c10f794e4005eec0cd9628ada13524b47b2a22c8/post-processor/digitalocean-import/post-processor.go#L202), as well.

**Review by nywilken — CHANGES_REQUESTED** [review]

**Comment by bhcleek** [comment]
@nywilken PTAL

Thank you for pointing out the post-processor. I should have done a more thorough search. I've updated it and the acceptance tests.

I also made a small refactor to use the godo `ClientOpt`s so that the user agent string will also include the godo version.

**Review by nywilken — APPROVED** [review]
> I also made a small refactor to use the godo ClientOpts so that the user agent string will also include the godo version

Good call. The updated changes look good! Thanks for the quick turnaround.

**Pull request merged**